notes/lecture17.tex

```diff
@@ -0,0 +1,189 @@
+        &= \frac{\partial g(0)}{\partial \alpha} \nonumber \\
+        &= d^\trans \nabla f(x^*) \nonumber
+    \end{align}
+    \ref{eqn:0} Because $x^*$ is a local minimum, $0 \le f(x^* + \alpha d) - f(x^*)$ for sufficiently small alpha. So, we can construct a sequence with only positive $\alpha$ that converges to $x^*$ such that each element $0 \le \frac{f(x^* + \alpha_n d) - f(x^*)}{\alpha_n}$ which implies that statement given that $f$ is locally differentiable.\\
+    \\
+    Since $d$ is arbitrary, this implies that $\nabla f(x^*) = 0$.
+    \item First we represent $f(x^* + \alpha d) - f(x^*)$ using the 2nd order Taylor expansion.
+    \begin{align*}
+        f(x^* + \alpha d) - f(x^*) &= \alpha \nabla f(x^*)^\trans d + \frac{\alpha^2}{2}d^\trans \nabla^2 f(x^*)d + O(\alpha^2) \nonumber\\
+        &= \frac{\alpha^2}{2} d^\trans \nabla^2 f(x^*) d + O(\alpha^2) \\
+    \end{align*}
+    Now we do the following
+    \begin{align*}
+        0 &\le \lim_{\alpha \to 0} \frac{f(x^* + \alpha d) - f(x^*)}{\alpha^2} \\
+        &= \lim_{\alpha \to 0} \frac{1}{2}d^\trans \nabla^2 f(x^*) d + \frac{O(\alpha^2)}{\alpha^2} \\
+        &= \frac{1}{2}d^\trans \nabla^2 f(x^*) d
+    \end{align*}
+    Because $d$ is arbitrary, this implies that $\nabla^2 f(x^*) \succeq 0$ (Positive semidefinite).
+    
+\end{enumerate}
+\end{proof}
+
+Note that $\nabla f(x^*) = 0$ alone does not imply $x^*$ is a local minimum. Even the necessary conditions $\nabla f(x^*) = 0$ and $\nabla^2 f(x^*) \succeq 0$ does not imply $x^*$ is a local minimum. This is because it could be that $\nabla^2 f(x^*) = 0$, but the 3rd order is not 0. For example in the 1d case, $x^* = 0$ for $f(x) = x^3$ satisfies these conditions, but is not a local minimum. Now, we will look at the actual sufficient conditions for a local minimum, but these conditions can only detect strict local minimum.
+
+\begin{proposition}[Sufficient conditions for strict local minimum]
+\propositionlabel{sufficient condition for strict local min}
+    Let $f\colon \R^n \to \R$ be twice continuously differentiable ($C^2$) over an open set $S$. Suppose $x \in S$ such that $\nabla f(x^*) = 0$ and $\nabla^2f(x) \succ 0$ (positive definite). Then, $x^*$ is a strict unconstrained local minimum.
+\end{proposition}
+
+\begin{proof}[Proof of \propositionref{sufficient condition for strict local min}.]
+Fix $d \in \R^n$. Note that $d^\trans \nabla^2 f(x^*) d \ge \lambda_{\text{min}} \|d\|^2$, where $\lambda_\text{min}$ is the smallest eigen value of $\nabla^2 f(x^*)$.
+\begin{align}
+    f(x^* + d) - f(x^*) &= \nabla f(x^*)^\trans d + \frac{1}{2}d^\trans \nabla^2 f(x^*)d + O(\|d\|^2) \label{eqn:1}\\
+    &\ge \frac{\lambda_{\text{min}}}{2} \|d\|^2 + O(\|d\|^2) \nonumber \\
+    &= (\frac{\lambda_{\text{min}}}{2} + \frac{O(\|d\|^2)}{\|d\|^2}) \|d\|^2 \nonumber\\
+    &> 0 \label{eqn:2}
+\end{align}
+\ref{eqn:1} 2nd Order Taylor expansion\\
```

Comment:
Same here. Please turn these into sentences. Also, put parantheses around the equation number, unless it's followed a noun like `Equation` or `Inequality`. So, after compiling it should show up as either `(6)`, or `Equality 6`.